src/shared/defaultAwsContextCommands.ts

```diff
@@ -280,18 +282,19 @@ export class DefaultAWSContextCommands {
      * regions are shown. Regions provided must exist in the available regions to be shown.
      */
     private async promptForRegion(regions?: string[], title?: string): Promise<string | undefined> {
-        const availableRegions = await this._regionProvider.getRegionData()
-        const regionItems = availableRegions
+        const partitionRegions = getRegionsForActiveCredentials(this._awsContext, this._regionProvider)
+
+        const regionsToShow = partitionRegions
```

Comment:
Is the concept of partitions hidden from the user? Would it be confusing that for a given credential I can only pick a subset without any explanation why? In reality I get they'll probably know - I'm just playing devils advocate

Reply:
The experience here parallels the web console, in that you can only see regions for the partition your account is based in. It would be a jarring experience to see and attempt to interact with regions that you'd get access related errors.

It seems like users working in a partition other than 'aws' typically require extra steps to get up and running (for example, setting region args), and would likely have had exposure to documentation related to their partition, including available regions.

Reply:
We may want to add the above to the docs and have a help button on the selector take them to the explanation.